addons/calendar/models/calendar_event.py

```diff
@@ -814,6 +814,51 @@ def copy(self, default=None):
             new_event.write({'partner_ids': [(Command.set(old_event.partner_ids.ids))]})
         return new_events
 
+    def action_unlink_event(self, attendee_id=None, recurrence=False):
+        """
+        Delete the event after displaying the delete wizard if necessary.
+
+        :param attendee_id: The ID of the attendee for the event
+        :param recurrence: Boolean indicating if the event is recurring
+
+        :return: Action to delete the event
+        """
+        if self.user_id._has_any_active_synchronization() or len(self.ids) > 1:
+            self.unlink()
+            return {
+                'type': 'ir.actions.act_url',
+                'target': 'self',
+                'url': '/odoo/calendar'
+            }
+
+        template_id = self.env['ir.model.data']._xmlid_to_res_id(
+            'calendar.calendar_template_delete_event', raise_if_not_found=False
+        )
```

Comment:
Why not use self.env.ref('calendar.calendar_template_delete_event', raise_if_not_found=False) ?

Reply:
We use `self.env.ref` to get views like list or form views. However, when we need to fetch a template ID from a data file, we use `self.env['ir.model.data']`. 
Our codebase fetches email templates using `self.env['ir.model.data']`.

Reply:
> We use `self.env.ref` to get views like list or form views. However, when we need to fetch a template ID from a data file, we use `self.env['ir.model.data']`. Our codebase fetches email templates using `self.env['ir.model.data']`.

That's not true: https://github.com/odoo/enterprise/blob/18.0/sale_subscription/models/sale_order.py#L1296-L1297
https://github.com/odoo/odoo/blob/18.0/addons/sale/models/sale_order.py#L1056
No need to use self.env['ir.model.data'] directly.

Reply:
oops!! my bad :eyes: